hs_core/views/utils.py

```diff
@@ -1304,21 +1354,37 @@ def create_folder(res_id, folder_path, migrating_resource=False):
     :return:
     """
     if __debug__:
-        assert(folder_path.startswith("data/contents/"))
+        assert folder_path.startswith("data/contents/")
 
     resource = hydroshare.utils.get_resource_by_shortkey(res_id)
     if resource.raccess.published and not migrating_resource:
         raise ValidationError("Folder creation is not allowed for a published resource")
     istorage = resource.get_irods_storage()
-    coll_path = os.path.join(resource.root_path, folder_path)
-
+    # reconstruct the folder_path by stripping each of the folders in the path and then joining them back
+    folders = [folder.strip() for folder in folder_path[len("data/contents/"):].split("/")]
+    folder_path = os.path.join(*folders)
+    coll_path = os.path.join(resource.file_path, folder_path)
     if not resource.supports_folder_creation(coll_path):
         raise ValidationError("Folder creation is not allowed here. "
                               "The target folder seems to contain aggregation(s)")
 
+    folder_path = coll_path[len(resource.file_path) + 1:]
+    # folder name can't be '.' or '..'
+    if any(["/./" in coll_path, "/../" in coll_path, coll_path.endswith("/."), coll_path.endswith("/..")]):
+        err_msg = f"Folder path ({folder_path}) contains invalid folder name(s)."
+        raise SuspiciousFileOperation(err_msg)
+
     # check for duplicate folder path
     if istorage.exists(coll_path):
-        raise ValidationError("Folder already exists")
+        raise ValidationError(f"Folder ({coll_path}) already exists")
+
+    for folder in folders:
+        if not ResourceFile.is_folder_name_valid(folder):
+            folder_banned_chars = ResourceFile.banned_symbols().replace('/', '')
+            err_msg = f"Folder name ({folder}) contains one more prohibited characters. "
```

Comment:
![image](https://user-images.githubusercontent.com/17934193/203153577-00ab480c-5a78-4e8e-bf07-208bfb1ff436.png)
when attempt to create a folder name that doesn't comply, the "error" message is shown behind the modal
either we should hide the modal after clicking the "create" button, or even better we should show the "Folder Creation Failed" feedback within the "New Folder modal" so that feedback is immediate

Reply:
fixed this with hiding the dialog on error.

Reply:
nice, looks good!
LMK when you're ready for a re-review

Reply:
@devincowan It's now ready for a 2nd round of review.  I tried to fix quite a bit of error message formatting issues which initially I was reluctant to do in this PR.  I think I now have the check for file/folder name validation for all requests involving file/folder operations.